azurerm/resource_arm_network_security_group.go

```diff
@@ -264,23 +313,31 @@ func expandAzureRmSecurityRules(d *schema.ResourceData) ([]network.SecurityRule,
 
 		name := data["name"].(string)
 		source_port_range := data["source_port_range"].(string)
+		source_port_ranges := data["source_port_ranges"].([]string)
```

Comment:
So the reason for the crash is because [`source_port_ranges` in the schema is a Set](https://github.com/terraform-providers/terraform-provider-azurerm/pull/882/files#diff-02a62c4182918d899d123e535c80ac12R70) - which means it needs to be cast to/from a Set (which can be of any type) like so:

````
sourcePortRanges := data["source_port_ranges"].(*schema.Set).List()
outputs := make([]string, 0)

for _, sourcePortRange := range sourcePortRanges {
  outputs = append(outputs, sourcePortRange.(string)
}
```

Hope that helps :)